examples/tti_example.py

```diff
@@ -1,59 +1,173 @@
+    parser.add_argument("-r", "--resultsdir", default="results",
+                        help="Directory containing results")
+    parser.add_argument("-p", "--plotdir", default="plots",
+                        help="Directory containing plots")
+    parser.add_argument("--max_bw", type=float, help="Maximum bandwith of the system")
+    parser.add_argument("--max_flops", type=float, help="Maximum FLOPS of the system")
+
+    args = parser.parse_args()
+
+    parameters = vars(args).copy()
+    del parameters["execmode"]
+    del parameters["resultsdir"]
+    del parameters["plotdir"]
+    del parameters["max_bw"]
+    del parameters["max_flops"]
+    del parameters["omp"]
+
+    parameters["dimensions"] = tuple(parameters["dimensions"])
+    parameters["spacing"] = tuple(parameters["spacing"])
+    parameters["cache_blocking"] = parameters["cache_blocking"] + [None]
+    parameters["compiler"] = compiler_registry[args.compiler](openmp=args.omp)
+
+    if args.execmode == "run":
+        run(**parameters)
+
+    if args.execmode == "bench":
+        if Benchmark is None:
+            raise ImportError("Could not find opescibench utility package.\n"
+                              "Please install from https://github.com/opesci/opescibench")
+
+        if parameters["auto_tuning"]:
+            parameters["auto_tuning"] = [True, False]
+
+        if parameters["cse"]:
+            parameters["cse"] = [True, False]
+
+        class TTIExecutor(Executor):
+            """Executor class that defines how to run TTI benchmark"""
+
+            def run(self, *args, **kwargs):
+                gflops, oi = run(*args, **kwargs)
+
+                self.register(gflops["kernel"], measure="gflops")
+                self.register(oi["kernel"], measure="oi")
+
+        bench = Benchmark(name="TTI", resultsdir=args.resultsdir, parameters=parameters)
+        bench.execute(TTIExecutor(), warmups=0)
+        bench.save()
+
+    if args.execmode == "plot":
+        if parameters["auto_tuning"]:
+            parameters["auto_tuning"] = [True, False]
+
+        if parameters["cse"]:
+            parameters["cse"] = [True, False]
+
+        bench = Benchmark(name="TTI", resultsdir=args.resultsdir, parameters=parameters)
+        bench.load()
+
+        oi_dict = {}
+        mflops_dict = {}
+
+        gflops = bench.lookup(params=parameters, measure="gflops")
+        oi = bench.lookup(params=parameters, measure="oi")
+
+        for key, gflops in gflops.items():
+            oi_value = oi[key]
+            key = dict(key)
+            label = "TTI, CSE: %s, AT: %s" % (key["cse"], key["auto_tuning"])
+            mflops_dict[label] = gflops * 1000
+            oi_dict[label] = oi_value
+
+        name = "TTI - Dimensions: %s, Spacing: %s.pdf" % \
+            (parameters["dimensions"], parameters["spacing"])
+
+        plotter = Plotter()
+        plotter.plot_roofline(
+            name, mflops_dict, oi_dict, args.max_bw, args.max_flops)
```

Comment:
I understand. I was just pointing out that if we want to advertize what we do well, it is better to split this examples into "one script , one purpose" so that people can easily find what they need without having to filter out half an example themselves